packages/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt

```diff
@@ -259,10 +261,12 @@ class ReactPlugin : Plugin<Project> {
 
     // We tell Android Gradle Plugin that inside /build/generated/autolinking/src/main/java there
     // are sources to be compiled as well.
-    project.extensions.getByType(AndroidComponentsExtension::class.java).apply {
-      onVariants(selector().all()) { variant ->
-        variant.sources.java?.addStaticSourceDirectory(
+    if (!isLibrary) {
```

Comment:
can we adjust the comment so that it's clear on why we don't need that for libraries?

Reply:
Resolving this.